### ISSUE TITLE
Another M1 build/link error - "Undefined symbols for architecture arm64" with openssl library

### ISSUE THREAD

**Issue by darrenfu** [facebookincubator/velox#3436]
### Problem description

*Since end of Nov, this build error started to pop up when make, and rebuild and run both via Command line and via Clion. It is not reproducible every time, but it has 50-70% chance to recur.*

E.g. issuing a make command:
`make clean && CPU_TARGET="arm64" make debug`

**Fun fact** Every time it rebuilds, it looks like some program is downloading the x86-64 openssl library with root user permission, and saved it at: `/opt/homebrew/Cellar/openssl@1.1/1.1.1d/`. 
```
➜  ls -lh /opt/homebrew/Cellar/openssl@1.1/
total 0
drwxr-xr-x  13 root  wheel   416B Dec  6 12:56 1.1.1d
drwxr-xr-x  14 dofu  staff   448B Dec  5 15:27 1.1.1s
```

Even if I try to download an arm64 version via `brew reinstall openssl@1.1`, and relink it to the arm64 version. It will get overridden after the rebuild.
```
➜  ls -lh /opt/homebrew/opt/openssl@1.1
lrwxr-xr-x  1 root  wheel    28B May 15  2020 /opt/homebrew/opt/openssl@1.1 -> ../Cellar/openssl@1.1/1.1.1d
```
➜  otool -vh /opt/homebrew/Cellar/openssl@1.1/1.1.1*/lib/libssl.1.1.dylib
/opt/homebrew/Cellar/openssl@1.1/1.1.1d/lib/libssl.1.1.dylib:
Mach header
      magic  cputype cpusubtype  caps    filetype ncmds sizeofcmds      flags
MH_MAGIC_64   X86_64        ALL  0x00       DYLIB    15       1648   NOUNDEFS DYLDLINK TWOLEVEL NO_REEXPORTED_DYLIBS
/opt/homebrew/Cellar/openssl@1.1/1.1.1s/lib/libssl.1.1.dylib:
Mach header
      magic  cputype cpusubtype  caps    filetype ncmds sizeofcmds      flags
MH_MAGIC_64    ARM64        ALL  0x00       DYLIB    17       1488   NOUNDEFS DYLDLINK TWOLEVEL NO_REEXPORTED_DYLIBS

I tried to add the following global variables into `~/.zshrc`, but it doesn't solve the problem.
```
export CPPFLAGS="-I/opt/homebrew/Cellar/openssl@1.1/1.1.1s/include"
export OPENSSL_ROOT_DIR=/opt/homebrew/Cellar/openssl@1.1/1.1.1s
export OPENSSL_LIBRARIES=/opt/homebrew/Cellar/openssl@1.1/1.1.1s/lib
export PKG_CONFIG_PATH="/opt/homebrew/Cellar/openssl@1.1/1.1.1s/lib/pkgconfig"
```

The only dirty workaround is to relink the openssl library directory. 
```
sudo rm -f  /opt/homebrew/opt/openssl@1.1 && ln -s /opt/homebrew/Cellar/openssl@1.1/1.1.1s /opt/homebrew/opt/openssl@1.1
```

Note: This is not the same issue of https://github.com/facebookincubator/velox/issues/3431.

Need some ultimate fix on this issue.

cc @kgpai @Yuhta 

### System information

Velox System Info v0.0.2
Commit: 7ed6f5f113b612353fc6b52236f57fef6a1f47cc
CMake Version: 3.25.0
System: Darwin-21.6.0
Arch: arm64
C++ Compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
C++ Compiler Version: 14.0.0.14000029
C Compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
C Compiler Version: 14.0.0.14000029
CMake Prefix Path: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk/usr;/opt/homebrew;/usr/local;/usr;/;/opt/homebrew/Cellar/cmake/3.25.0;/usr/local;/usr/X11R6;/usr/pkg;/opt;/sw;/opt/local
  

I'm using XCode 14's clang. 

### CMake log

```bash
ld: warning: ignoring file /opt/homebrew/opt/openssl@1.1/lib/libssl.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /opt/homebrew/opt/openssl@1.1/lib/libcrypto.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64


https://gist.github.com/darrenfu/0e1916c85ee5a27e3f7230c8955241ec
```

**Comment by darrenfu** [comment]
I was able to find an ultimate solution for this issue.  
TL;DR: DO NOT INSTALL `homebrew` IN YOUR DEFAULT LOCATION (for Apple M1 users, it's under `/opt/homebrew/`). Otherwise, you'll get issues with openSSL libraries just like this one. 

**Detailed steps**:
1. Uninstall the homebrew. [Here](https://github.com/homebrew/install#uninstall-homebrew) is the instruction.
2. Next is a cleaner solution to reinstall homebrew:
    1. Clone homebrew to your home directory: `git clone https://github.com/Homebrew/brew ~/homebrew`
    3. Add brew binary folders to the PATH environmental variable (echo $SHELL to check what shell you're using):
        1. Z-Shell: `echo 'export PATH="$HOME/homebrew/bin:$HOME/homebrew/sbin:$PATH"' >> ~/.zshrc`, then `source ~/.zshrc`
        2. Bash shell: `echo 'export PATH="$HOME/homebrew/bin:$HOME/homebrew/sbin:$PATH"' >> ~/.bash_profile`, then `source ~/.bash_profile`
     4. Run `brew update`; then use run `./scripts/setup-macos.sh`

**Comment by ulysses-you** [comment]
@darrenfu thank you for the information ! Add more one tip if there is two version brew.

Specifec the target arm64 brew home when setup dependencies:
```
CPU_TARGET="arm64" INSTALL_PREFIX=/opt/homebrew ./scripts/setup-macos.sh
```

**Comment by chym1303** [comment]
I encountered the same issue, it was caused by miniforge. And I switch to a new conda env, delete the libssl in conda, the fix the problem.
